### PR TITLE
94 implement sapexauthservice project

### PR DESCRIPTION
**Pull request by nurmukhammadsunatullaev**
Created SAPexAuthService project!

### CODE REVIEW

**Comment by nurmukhammadsunatullaev** [comment]
[Implement SAPexAuthService project](https://trello.com/c/lrzbAYQz/94-implement-sapexauthservice-project)

File: BusinessLogicLayer/Services/LanguageService.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public async Task<bool> AddLanguageAsync(LanguageDtoModel languageDto)
             {
                 LanguageEntityModel languageEM = profile.mapToEM(languageDto);
                 await Task.Run(() => unitOfWork.Languages.CreateAsync(languageEM));
-                unitOfWork.SaveAsync();
+                await Task.Run(() => unitOfWork.SaveAsync());
```
Comment (dnavitski) [inline BusinessLogicLayer/Services/LanguageService.cs]: Please remove Task.Run()

File: BusinessLogicLayer/Services/SandboxService.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ public async Task<bool> AddSandboxAsync(SandboxDtoModel sandboxDto)
             {
                 SandboxEntityModel sandbox = profile.mapToEM(sandboxDto);
                 await Task.Run(() => unitOfWork.Sandboxes.CreateAsync(sandbox));
-                unitOfWork.SaveAsync();
+                await Task.Run(() => unitOfWork.SaveAsync());
```
Comment (dnavitski) [inline BusinessLogicLayer/Services/SandboxService.cs]: Please remove Task.Run()

File: DataAccessLayer/Repositories/UserRepository.cs [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
+
 namespace DataAccessLayer.Repositories
 {
     public class UserRepository : Repository<UserEntityModel>, IUserRepository
     {
         public UserRepository(AppDbContext context)
             : base(context)
         { }
+
+        //public override async Task<IEnumerable<UserEntityModel>>  FindByConditionAsync(Expression<Func<UserEntityModel, bool>> expression)
```
Comment (dnavitski) [inline DataAccessLayer/Repositories/UserRepository.cs]: Please remove commented code

File: SAPex/Controllers/FakeDB/StatusViewModel.json [hunk 4]
```diff
@@ -1 +1,42 @@
-[{"Name":"Draft","Id":"be7ba319-03a4-4ecd-a13c-ae1cbfffd7cc"},{"Name":"Rejected","Id":"ea6da7df-72cc-41a1-9fd7-70a373b0af94"},{"Name":"Test","Id":"2ac99400-09a2-4872-bd69-0f555286267e"},{"Name":"Need Verification","Id":"c08eab63-a92d-4f3a-ba3f-a8de3acd9513"},{"Name":"Need Recruiter","Id":"30ae8876-359f-4c4a-88c2-8fbc0f66b388"},{"Name":"Approved","Id":"a1bec403-45e6-44ab-b78c-82d4d083ec19"},{"Name":"Questionable","Id":"3a34496a-9141-4c2a-8da4-a6e71a2b0449"},{"Name":"Interview Soft","Id":"94d2d509-118d-4e59-984e-3cda6019fc76"},{"Name":"Interview Rejected","Id":"e2732ba8-e870-4a28-8fb7-9e6cd129a770"},{"Name":"Interview Tech","Id":"13e82ea7-e78b-4185-963c-a2f23cf866e3"}]
\ No newline at end of file
+[
```
Comment (dnavitski) [inline SAPex/Controllers/FakeDB/StatusViewModel.json:1]: Please revert file changes.

File: SAPex/appsettings.json [hunk 5]
```diff
@@ -11,10 +11,11 @@
   "AppSettings": {
     "Secret": "PDv7DrqznYL6nv7DrqzjnQYO9JxIsWdcjnQYL6nu0f",
     "ExpDate": 1,
-    "ExpDay": 7
+    "ExpMonth": 7
   },
   "ConnectionStrings": {
-    "DefaultConnection": "Server=(LocalDB)\\MSSQLLocalDB;Database=SAPExDb;Trusted_Connection=True;MultipleActiveResultSets=true;"
+    "DefaultConnection": "Server=127.0.0.1,1433;Database=SAPExDb;User Id=SA;Password=MyPass@word;"
+    //"DefaultConnection": "Server=(LocalDB)\\MSSQLLocalDB;Database=SAPExDb;Trusted_Connection=True;MultipleActiveResultSets=true;"
```
Comment (dnavitski) [inline SAPex/appsettings.json]: Please comment on your personal connection string and uncomment the previous connection string.

**Review by dnavitski — CHANGES_REQUESTED** [review]

File: BusinessLogicLayer/Services/LanguageService.cs (hunk 1 above)
Comment (nurmukhammadsunatullaev) [reply BusinessLogicLayer/Services/LanguageService.cs]: Dear mentor but it is not my controller

File: BusinessLogicLayer/Services/SandboxService.cs (hunk 2 above)
Comment (nurmukhammadsunatullaev) [reply BusinessLogicLayer/Services/SandboxService.cs]: this is too

File: BusinessLogicLayer/Services/LanguageService.cs (hunk 1 above)
Comment (dnavitski) [reply BusinessLogicLayer/Services/LanguageService.cs]: await Task.Run(() => unitOfWork.SaveAsync());
this code was changed in your branch, I am sorry, you need update it

File: BusinessLogicLayer/Services/SandboxService.cs (hunk 2 above)
Comment (dnavitski) [reply BusinessLogicLayer/Services/SandboxService.cs]: this should be updated by yourself too
await Task.Run(() => unitOfWork.SaveAsync());

**Review by dnavitski — APPROVED** [review]

**Review by dnovitsky — APPROVED** [review]

**Pull request merged**